test/bmc/bmc_api_test.rb

```diff
@@ -64,10 +64,10 @@ def test_api_bmc_setup_returns_new_ipmi_proxy_given_ipmitool
     api = Proxy::BMC::Api.new!
     Proxy::BMC::Plugin.settings.stubs(:bmc_default_provider).returns('freeipmi')
     Proxy::BMC::Plugin.settings.stubs(:provider_log_level).returns(nil)
-    auth = Object.new
-    auth.stubs(:provided?).returns(true)
-    auth.stubs(:basic?).returns(true)
-    auth.stubs(:credentials).returns('username','password')
+    auth = mock()
+    auth.expects(:provided?).returns(true)
+    auth.expects(:basic?).returns(true)
+    auth.expects(:credentials).returns('username','password')
```

Comment:
My understanding is that this test verifies that ```ipmitool``` is being used when ```bmc_provider``` is set appropriately. If this is the case, I would stick with stubs for authorization (not really important for this test whether auth calls have been made), but would probably use a mock for ```find_ipmi_provider```, together with verifying parameters passed to it...